### PR TITLE
Switch to Github Actions and push images to ECR

### PR DESCRIPTION
**Pull request by guilgaly**

### CODE REVIEW

File: kubernetes/02-deployment.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
     spec:
       containers:
         - name: demostore
-          image: eu.gcr.io/production-243513/demostore:master
+          image: 049894548288.dkr.ecr.us-east-1.amazonaws.com/demostore:master
```
Comment (Isammoc) [inline kubernetes/02-deployment.yml]: As the change aims to target AWS, there is no more Kubernetes involved.

Non need to change this file (or you may delete it)

File: Jenkinsfile [hunk 2]
```diff
@@ -36,7 +36,7 @@ spec:
       steps {
         container('docker') {
           script {
-            docker.build('eu.gcr.io/production-243513/demostore:' + branch).push()
+            docker.build('049894548288.dkr.ecr.us-east-1.amazonaws.com/demostore:' + branch).push()
```
Comment (Isammoc) [inline Jenkinsfile]: You need to add credentials as it was on our [FrontLine repository](https://github.com/gatling/frontline-cloud/blob/9601a70dd868f66f91b8ac9b4247a821f700f3c0/.jenkins/release.Jenkinsfile#L117-L126)

But better to convert the whole thing in github actions as it is what we want finally.

**Review by Isammoc — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Cache Maven
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+
+      - name: Login to Amazon ECR
+        id: ecr-login
+        uses: aws-actions/amazon-ecr-login@v1
```
Comment (Isammoc) [inline .github/workflows/build.yml:51]: This step required that AWS CLI is configured.
https://github.com/gatling/frontline-cloud/blob/main/.github/workflows/release.yml#L27-L32

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          ECR_REPOSITORY: demostore
+          DEMOSTORE_VERSION: ${{ env.GITHUB_REF_SLUG }} # branch name
+        run: |
+          docker build \
+            --tag $ECR_REGISTRY/$ECR_REPOSITORY:$DEMOSTORE_VERSION \
+            --build-arg FRONTLINE_VERSION="$FRONTLINE_VERSION" \
+            --file Dockerfile
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$DEMOSTORE_VERSION
+          docker tag $ECR_REGISTRY/$ECR_REPOSITORY:$DEMOSTORE_VERSION $ECR_REGISTRY/$ECR_REPOSITORY:latest
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:latest
```
Comment (Isammoc) [inline .github/workflows/build.yml]: Split into 2 steps

 * docker build: for all branches
 * docker push: only for `main`

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Docker
+        env:
+          ECR_REGISTRY: ${{ steps.ecr-login.outputs.registry }}
+          ECR_REPOSITORY: demostore
+          DEMOSTORE_VERSION: ${{ env.GITHUB_REF_SLUG }} # branch name
+        run: |
+          docker build \
+            --tag $ECR_REGISTRY/$ECR_REPOSITORY:$DEMOSTORE_VERSION \
+            --build-arg FRONTLINE_VERSION="$FRONTLINE_VERSION" \
```
Comment (Isammoc) [inline .github/workflows/build.yml]: copy pasta!

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        id: ecr-login
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v3.x
+
+      - name: Docker
+        env:
+          ECR_REGISTRY: ${{ steps.ecr-login.outputs.registry }}
+          ECR_REPOSITORY: demostore
```
Comment (Isammoc) [inline .github/workflows/build.yml:59]: Is it the name in ECR? (I don't know what ops put)

File: .github/workflows/build.yml (hunk 4 above)
Comment (guilgaly) [reply .github/workflows/build.yml]: Done.

**Review by Isammoc — APPROVED** [review]

**Pull request merged**